src/main/java/Rica.java

```diff
@@ -0,0 +1,154 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class Rica {
+    private static final String LINE = "____________________________________________________________";
+    private static final String LIST_TRIGGER = "list";
+    private static final String MARK_TRIGGER = "mark";
+    private static final String UNMARK_TRIGGER = "unmark";
+    private static final String BYE_TRIGGER = "bye";
+    private static final String ADD_PHRASE = " New task I'll remember: ";
+    private static final String BYE_PHRASE = " Leaving so soon? Come back anytime, I'll be happy to help!";
+    private static ArrayList<Task> pastTasks = new ArrayList<>();
+
+    private static ArrayList<Task> getPastTexts() {
+        return Rica.pastTasks;
+    }
+
+    /**
+     * Adds a new task to the task list with the given description
+     *
+     * @param description Description of new task being added
+     */
+    private static void addTask(String description) {
+        if (description.isBlank()) {
+            printlnWithIndent(" Did you forget something? Don't think there's any task with no name");
+            return;
+        }
+        Task newTask = new Task(description);
+        Rica.getPastTexts().add(newTask);
+        printlnWithIndent(Rica.ADD_PHRASE + description);
+    }
+
+    /**
+     * Mark a given task in the task list as done
+     *
+     * @param indexOfTask Index of given task in the task list
+     * @return Task object representing the desired task being marked as done
+     */
+    private static Task markDone(int indexOfTask) {
+        if (indexOfTask < 0 || indexOfTask >= Rica.getPastTexts().size()) {
+            printlnWithIndent(" You alright? I can't mark a task that doesn't exist as done xD");
+            return null;
+        }
+        Task selectedTask = Rica.getPastTexts().get(indexOfTask);
+        if (selectedTask.getIsDone()) {
+            printlnWithIndent(" Take a break maybe? Alright marked as done my friend:");
+            printlnWithIndent("    " + selectedTask);
+            return selectedTask;
+        }
+        Rica.getPastTexts().remove(indexOfTask);
+        selectedTask = selectedTask.setDone();
+        Rica.getPastTexts().add(indexOfTask, selectedTask);
+        printlnWithIndent(" Shall remember that this task as done:");
+        printlnWithIndent("    " + selectedTask);
+        return selectedTask;
+    }
+
+    /**
```

Comment:
Comments can be evenly placed into the individual lines of code for better clarity on what each line of code does